server/tests-py/test_jwt.py

```diff
@@ -42,23 +42,33 @@ def mk_claims(conf, claims):
 class TestJWTBasic():
 
     def test_jwt_valid_claims_success(self, hge_ctx, endpoint):
-        self.claims['https://hasura.io/jwt/claims'] = mk_claims(hge_ctx.hge_jwt_conf, {
-            'x-hasura-user-id': '1',
-            'x-hasura-allowed-roles': ['user', 'editor'],
-            'x-hasura-default-role': 'user'
+        claims = mk_claims(hge_ctx.hge_jwt_conf, {
+                'x-hasura-user-id': '1',
+                'x-hasura-allowed-roles': ['user', 'editor'],
+                'x-hasura-default-role': 'user'
         })
+        claims_namespace_path = hge_ctx.hge_jwt_claims_ns_path
+        if claims_namespace_path is None:
+            self.claims['https://hasura.io/jwt/claims'] = claims
+        elif claims_namespace_path == "$.hasuraClaims":
+            self.claims['hasuraClaims'] = claims
         token = jwt.encode(self.claims, hge_ctx.hge_jwt_key, algorithm='RS512').decode('utf-8')
         self.conf['headers']['Authorization'] = 'Bearer ' + token
         self.conf['url'] = endpoint
         self.conf['status'] = 200
-        check_query(hge_ctx, self.conf, add_auth=False)
+        check_query(hge_ctx, self.conf, add_auth=False,claims_namespace_path=claims_namespace_path)
 
     def test_jwt_invalid_role_in_request_header(self, hge_ctx, endpoint):
-        self.claims['https://hasura.io/jwt/claims'] = mk_claims(hge_ctx.hge_jwt_conf, {
+        claims = mk_claims(hge_ctx.hge_jwt_conf, {
             'x-hasura-user-id': '1',
             'x-hasura-allowed-roles': ['contractor', 'editor'],
             'x-hasura-default-role': 'contractor'
         })
+        claims_namespace_path = hge_ctx.hge_jwt_claims_ns_path
+        if claims_namespace_path is None:
+            self.claims['https://hasura.io/jwt/claims'] = claims
+        elif claims_namespace_path == "$.hasuraClaims":
+            self.claims['hasuraClaims'] = claims
```

Comment:
This logic is repeated lot of places in the same file. If possible, can we abstract this into a new function and reuse it?

Reply:
Yes, I have done it in the new commit